coordinator/persistence-cassandra-4.1/src/main/java/io/stargate/db/cassandra/impl/interceptors/DefaultQueryInterceptor.java

```diff
@@ -0,0 +1,301 @@
+    for (EventListener listener : listeners) {
+      listener.onUp(nativeAddress.getAddress(), nativeAddress.getPort());
+    }
+  }
+
+  private void applyState(
+      InetAddressAndPort endpoint,
+      ApplicationState state,
+      VersionedValue value,
+      EndpointState epState) {
+    switch (state) {
+      case RELEASE_VERSION:
+        updatePeer(endpoint, value.value, StargatePeerInfo::setReleaseVersion);
+        break;
+      case DC:
+        updatePeer(endpoint, value.value, StargatePeerInfo::setDataCenter);
+        break;
+      case RACK:
+        updatePeer(endpoint, value.value, StargatePeerInfo::setRack);
+        break;
+      case RPC_ADDRESS:
+        try {
+          updatePeer(
+              endpoint, InetAddress.getByName(value.value), StargatePeerInfo::setNativeAddress);
+        } catch (UnknownHostException e) {
+          throw new RuntimeException(e);
+        }
+        break;
+      case NATIVE_ADDRESS_AND_PORT:
+        try {
+          InetAddressAndPort address = InetAddressAndPort.getByName(value.value);
+          updatePeer(endpoint, address.getAddress(), StargatePeerInfo::setNativeAddress);
+          updatePeer(endpoint, address.getPort(), StargatePeerInfo::setNativePort);
+        } catch (UnknownHostException e) {
+          throw new RuntimeException(e);
+        }
+        break;
+      case HOST_ID:
+        updatePeer(endpoint, UUID.fromString(value.value), StargatePeerInfo::setHostId);
+        break;
+      case RPC_READY:
+        notifyRpcChange(endpoint, epState.isRpcReady());
+        break;
+      default:
+        // ignore other states
+    }
+  }
+
+  private <V> void updatePeer(
+      InetAddressAndPort endpoint, V value, BiConsumer<StargatePeerInfo, V> updater) {
+
+    if (!endpoint.equals(FBUtilities.getBroadcastAddressAndPort())) {
+      StargatePeerInfo peer =
+          StargateSystemKeyspace.instance
+              .getPeers()
+              .computeIfAbsent(endpoint, StargatePeerInfo::new);
+      updater.accept(peer, value);
+    }
+  }
+
+  private void notifyDown(InetAddressAndPort endpoint) {
+    InetAddressAndPort nativeAddress = getNativeAddress(endpoint);
+    for (EventListener listener : listeners) {
+      listener.onDown(nativeAddress.getAddress(), nativeAddress.getPort());
+    }
+  }
+
+  private InetAddressAndPort getNativeAddress(InetAddressAndPort endpoint) {
+    try {
+      return InetAddressAndPort.getByName(StorageService.instance.getNativeaddress(endpoint, true));
+    } catch (UnknownHostException e) {
+      // That should not happen, so log an error, but return the
+      // endpoint address since there's a good change this is right
+      logger.error("Problem retrieving RPC address for {}", endpoint, e);
+      return endpoint;
+    }
+  }
+
+  private static boolean isStargateNode(EndpointState epState) {
+    VersionedValue value = epState.getApplicationState(ApplicationState.X10);
```

Comment:
Sure would be nice to know why, how and so on. :-/

It seems to control joining cluster (or not), but seems like a misnomer.
At first thought it might be related to "dev mode" but I suspect that's not it.
But not sure what is.